tests/test_pipeline_rebuild_guard.py

```
@@ -0,0 +1,296 @@
+    # defer to the canonical-audit detector (return None).
+    p.step_results = {
+        "absorb": AbsorbStepResult(success=True, promoted_slugs=[]),
+        "moc": MocStepResult(success=True, updated=False, changed_files=[]),
+    }
+    assert p._local_indexed_change_reason() is None
+
+
+def test_autopilot_passes_processed_article_reason():
+    """Autopilot's _run_knowledge_index_refresh only runs after an
+    article passed quality — it must always declare local change."""
+    import ovp_pipeline.autopilot.daemon as daemon
```

Ruff (0.15.12)

[warning] Use `from ovp_pipeline.autopilot import daemon` in lieu of alias

Replace with `from ovp_pipeline.autopilot import daemon`

(PLR0402)